examples/platform-specific/cc26xx/cc26xx-ble-client-demo/client.c

```diff
@@ -0,0 +1,134 @@
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE
+ * COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * \file
+ *    A simple IPv6-over-BLE UDP-client.
+ *
+ * \author
+ *    Michael Spoerk <michael.spoerk@tugraz.at>
+ */
+/*---------------------------------------------------------------------------*/
+#include "contiki.h"
+#include "contiki-lib.h"
+#include "contiki-net.h"
+
+#include "net/ipv6/uip-icmp6.h"
+
+#include "ble-hal.h"
+
+#include <string.h>
+#include <stdio.h>
+/*---------------------------------------------------------------------------*/
+#define SERVER_IP               "::"
+#define CLIENT_PORT           61617
+#define SERVER_PORT           61616
+
+#define PING_TIMEOUT              (CLOCK_SECOND / 4)
+#define CLIENT_SEND_INTERVAL      (CLOCK_SECOND * 1)
+
+#define UDP_LEN_MAX           255
+/*---------------------------------------------------------------------------*/
+static uip_ipaddr_t server_addr;
+static struct uip_icmp6_echo_reply_notification icmp_notification;
+static uint8_t echo_received;
+static struct uip_udp_conn *conn;
+
+static struct etimer timer;
+static char buf[UDP_LEN_MAX];
+static uint16_t packet_counter;
+/*---------------------------------------------------------------------------*/
+PROCESS(ipv6_ble_client_process, "IPv6 over BLE - client process");
+AUTOSTART_PROCESSES(&ipv6_ble_client_process);
+/*---------------------------------------------------------------------------*/
+void
+icmp_reply_handler(uip_ipaddr_t *source, uint8_t ttl,
+                   uint8_t *data, uint16_t datalen)
+{
+  if(uip_ip6addr_cmp(source, &server_addr)) {
+    printf("echo response received\n");
+    echo_received = 1;
+  }
+}
+/*---------------------------------------------------------------------------*/
+static void
+tcpip_handler(void)
+{
+  char data[UDP_LEN_MAX];
+  if(uip_newdata()) {
+    strncpy(data, uip_appdata, uip_datalen());
+    data[uip_datalen()] = '\0';
+    printf("rec. message: <%s>\n", data);
+  }
+}
+/*---------------------------------------------------------------------------*/
+static void
+timeout_handler(void)
+{
+  sprintf(buf, "Hello server %04u!", packet_counter);
+  printf("send message: <%s>\n", buf);
+  uip_udp_packet_send(conn, buf, strlen(buf));
+  packet_counter++;
+}
+/*---------------------------------------------------------------------------*/
+PROCESS_THREAD(ipv6_ble_client_process, ev, data)
+{
+  PROCESS_BEGIN();
+  printf("CC26XX-IPv6-over-BLE client started\n");
```

Comment:
Shouldn't this whole example move outside of `platform_specific`? Is it really cc26xx specific? As BLEach is portable, I would like to see a non-cc26xx example. Note that generally speaking, we platform-specific examples do not receive the same level of attention in terms of maintenance, testing etc.

BTW this will need a compile test (as detailed in a previous comment)